File: Program/ActEditor/src/pl/wroc/pwr/student/acteditor/parsing/xsd/Composition.java
package pl.wroc.pwr.student.acteditor.parsing.xsd;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Composition implements Element {
//	private List<Element> elements = new ArrayList<Element>();
	private List elements = Collections.synchronizedList(new ArrayList());
	private String description = "";
	private String name = "";
	private String type = "";
	
	public Composition(String name, String type) {
		this.name = name;
		this.type = type;
	}

	@Override
	public void add(Element element) {
		elements.add(element);
	}

	@Override
	public void remove(Element element) {
		elements.remove(element);
	}

	@Override
	public Element get(int index) {
		return (Element)elements.get(index);
	}

	@Override
	public List getElements() {
//		for(Element e : elements) {
//			e.getElements();
//		}
		return elements;
	}

	@Override
	public String getDescription() {
		return description;
	}

	@Override
	public void setDescription(String description) {
		this.description = description;
	}
	
	public String toString() {
		String result = name + " " + description;
		for(Object e : elements) {
			result += "\n\t" + e.toString();
		}
		return result;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getType() {
		return type;
	}

	public void setType(String type) {
		this.type = type;
	}

}

File: Program/ActEditor/bin/pl/wroc/pwr/student/acteditor/parsing/xsd/Composition.js
Clazz.declarePackage ("pl.wroc.pwr.student.acteditor.parsing.xsd");
Clazz.load (["pl.wroc.pwr.student.acteditor.parsing.xsd.Element", "java.util.ArrayList", "$.Collections"], "pl.wroc.pwr.student.acteditor.parsing.xsd.Composition", null, function () {
c$ = Clazz.decorateAsClass (function () {
this.elements = null;
this.description = "";
this.name = "";
this.type = "";
Clazz.instantialize (this, arguments);
}, pl.wroc.pwr.student.acteditor.parsing.xsd, "Composition", null, pl.wroc.pwr.student.acteditor.parsing.xsd.Element);
Clazz.prepareFields (c$, function () {
this.elements = java.util.Collections.synchronizedList ( new java.util.ArrayList ());
});
Clazz.makeConstructor (c$, 
function (name, type) {
this.name = name;
this.type = type;
}, "~S,~S");
Clazz.overrideMethod (c$, "add", 
function (element) {
this.elements.add (element);
}, "pl.wroc.pwr.student.acteditor.parsing.xsd.Element");
Clazz.overrideMethod (c$, "remove", 
function (element) {
this.elements.remove (element);
}, "pl.wroc.pwr.student.acteditor.parsing.xsd.Element");
Clazz.overrideMethod (c$, "get", 
function (index) {
return this.elements.get (index);
}, "~N");
Clazz.overrideMethod (c$, "getElements", 
function () {
return this.elements;
});
Clazz.overrideMethod (c$, "getDescription", 
function () {
return this.description;
});
Clazz.overrideMethod (c$, "setDescription", 
function (description) {
this.description = description;
}, "~S");
Clazz.defineMethod (c$, "toString", 
function () {
var result = this.name + " " + this.description;
for (var e, $e = this.elements.iterator (); $e.hasNext () && ((e = $e.next ()) || true);) {
result += "\n\t" + e.toString ();
}
return result;
});
Clazz.overrideMethod (c$, "getName", 
function () {
return this.name;
});
Clazz.overrideMethod (c$, "setName", 
function (name) {
this.name = name;
}, "~S");
Clazz.overrideMethod (c$, "getType", 
function () {
return this.type;
});
Clazz.overrideMethod (c$, "setType", 
function (type) {
this.type = type;
}, "~S");
});
